nats.go

```diff
@@ -1266,6 +1269,15 @@ func (nc *Conn) processExpectedInfo() error {
 		return err
 	}
 
+	if nc.Opts.Nkey != "" && nc.info.Nonce == "" {
+		return ErrNkeysNoSupported
+	}
+
+	// Check if we have an nkey but no signature callback defined.
```

Comment:
I meant this one should be moved in Options.Connect() itself.

Reply:
ok got it.